libraries/ESP8266WiFiMesh/src/FloodingMesh.cpp

```diff
@@ -0,0 +1,503 @@
+  
+  String messageID = generateMessageID();
+
+  // Remove getEspnowMeshBackend().getMeshName() from the broadcastMetadata below to broadcast to all ESP-NOW nodes regardless of MeshName.
+  String targetMeshName = getEspnowMeshBackend().getMeshName();
+
+  broadcastKernel(targetMeshName + String(broadcastMetadataDelimiter()) + messageID + String(broadcastMetadataDelimiter()) + message);
+}
+
+void FloodingMesh::broadcastKernel(const String &message)
+{
+  getEspnowMeshBackend().broadcast(message);
+}
+
+void FloodingMesh::setBroadcastReceptionRedundancy(uint8_t redundancy) 
+{ 
+  assert(redundancy < 255);
+  _broadcastReceptionRedundancy = redundancy;
+}
+uint8_t FloodingMesh::getBroadcastReceptionRedundancy() { return _broadcastReceptionRedundancy; }
+
+void FloodingMesh::encryptedBroadcast(const String &message)
+{
+  assert(message.length() <= maxEncryptedMessageSize());
+
+  String messageID = generateMessageID();
+  
+  encryptedBroadcastKernel(messageID + String(broadcastMetadataDelimiter()) + message);  
+}
+
+void FloodingMesh::encryptedBroadcastKernel(const String &message)
+{
+  getEspnowMeshBackend().attemptAutoEncryptingTransmission(message);
+}
+
+void FloodingMesh::clearMessageLogs()
+{
+  _messageIDs.clear();
+  std::queue<messageQueueElementType>().swap(_messageIdOrder);
+}
+
+void FloodingMesh::clearForwardingBacklog()
+{
+  _forwardingBacklog.clear();
+}
+ 
+void FloodingMesh::setMessageHandler(messageHandlerType messageHandler) { _messageHandler = messageHandler; }
+FloodingMesh::messageHandlerType FloodingMesh::getMessageHandler() { return _messageHandler; }
+
+void FloodingMesh::setOriginMac(uint8_t *macArray)
+{
+  std::copy_n(macArray, 6, _originMac);
+}
+
+String FloodingMesh::getOriginMac() { return macToString(_originMac); }
+uint8_t *FloodingMesh::getOriginMac(uint8_t *macArray)
+{
+  std::copy_n(_originMac, 6, macArray);
+  return macArray;
+}
+
+uint32_t FloodingMesh::maxUnencryptedMessageSize()
+{
+  return getEspnowMeshBackend().getMaxMessageLength() - MESSAGE_ID_LENGTH - (getEspnowMeshBackend().getMeshName().length() + 1); // Need room for mesh name + delimiter
+}
+
+uint32_t FloodingMesh::maxEncryptedMessageSize()
+{
+  // Need 1 extra delimiter character for maximum metadata efficiency (makes it possible to store exactly 18 MACs in metadata by adding an extra transmission)
+  return getEspnowMeshBackend().getMaxMessageLength() - MESSAGE_ID_LENGTH - 1;
+}
+ 
+void FloodingMesh::setMessageLogSize(uint16_t messageLogSize) 
+{ 
+  assert(messageLogSize >= 1);
+  _messageLogSize = messageLogSize; 
+}
+uint16_t FloodingMesh::messageLogSize() { return _messageLogSize; }
+
+void FloodingMesh::setBroadcastMetadataDelimiter(char broadcastMetadataDelimiter) 
+{ 
+  // Using HEX number characters as a delimiter is a bad idea regardless of broadcast type, since they are always in the broadcast metadata
+  assert(broadcastMetadataDelimiter < 48 || 57 < broadcastMetadataDelimiter);
```

Comment:
I left a comment above that I intended as an explanation of the situation (I'm checking for HEX characters), but reading it now I realize it may be read as a general statement. I'll rework this.